source/extensions/filters/http/rate_limit_quota/filter.cc

```diff
@@ -155,54 +155,39 @@ Http::FilterHeadersStatus RateLimitQuotaFilter::decodeHeaders(Http::RequestHeade
                           StreamInfo::CoreResponseFlag::ResponseFromCacheFilter);
 }
 
-void RateLimitQuotaFilter::createMatcher() {
-  RateLimitOnMatchActionContext context;
-  Matcher::MatchTreeFactory<Http::HttpMatchingData, RateLimitOnMatchActionContext> factory(
-      context, factory_context_.serverFactoryContext(), visitor_);
-  if (config_->has_bucket_matchers()) {
-    matcher_ = factory.create(config_->bucket_matchers())();
-  }
-}
-
 // TODO(tyxia) Currently request matching is only performed on the request
 // header.
 absl::StatusOr<Matcher::ActionPtr>
 RateLimitQuotaFilter::requestMatching(const Http::RequestHeaderMap& headers) {
   // Initialize the data pointer on first use and reuse it for subsequent
   // requests. This avoids creating the data object for every request, which
   // is expensive.
-  if (data_ptr_ == nullptr) {
-    if (callbacks_ != nullptr) {
-      data_ptr_ = std::make_unique<Http::Matching::HttpMatchingDataImpl>(callbacks_->streamInfo());
-    } else {
+  if (!data_ptr_) {
```

Comment:
i actually prefer `data_ptr == nullptr` style throughout this file. That is more readable for pointer type.